lib/core/Channel.js

```diff
@@ -0,0 +1,88 @@
+const ResultsParser = require("../parser/contents");
+const Simplify = require("../parser/simplify");
+const SimpleVideo = require("./SimpleVideo");
+
+class Channel {
+    #page;
+    #session;
+    constructor(session, data) {
+        this.#session = session;
+        this.#page = ResultsParser.parseResponse(data);
+    }
+
+    get title() {
+        return this.metadata.title || '';
+    }
+
+    get description() {
+        return this.metadata.description || '';
+    }
+
+    /**
+     * @type {import('../parser/contents/ChannelMetadata')}
+     */
+    get metadata() {
+        return this.#page.metadata;
+    }
+
+    /**
+     * 
+     * @returns {import('../parser/contents/Tab')[]}
+     */
+    getTabs() {
+        return Simplify.matching({
+            type: Simplify.matching(/^Tab$/),
+        }).runOn(this.#page);
+    }
+
+    /**
+     * 
+     * @param {string} name 
+     * @returns {import('../parser/contents/Tab')}
+     */
+    getTab(name) {
+        return Simplify.matching({
+            type: Simplify.matching(/^Tab$/),
+            title: Simplify.matching(new RegExp('^' + name + '$', 'i')),
+        }).runOn(this.#page)[0];
+    }
+
+    #getHomePageVideos() {
+        const tab = this.getTab('Home');
+        /**
+         * @type {import('../parser/contents/Shelf')[]}
+         */
+        const shelfs = Simplify.matching({
+            type: Simplify.matching(/^Shelf$/),
+        }).runOn(tab);
+        return shelfs.map(shelf => {
+            return {
+                title: shelf.title.toString(),
+                content: Simplify.matching({
+                    type: Simplify.matching(SimpleVideo.regex),
+                }).runOn(shelf).map(video => new SimpleVideo(video)),
+            }
+        });
+    }
+
+    /**
+     * @note home_page only returns videos!
+     * XXX: should some other API be made available to expose the content of the channel
```

Comment:
I think advanced users should just use the Simplify API (which I will still document) to get the other types of content on the home page

Maybe the `home_page` should also expose the channel trailer, but that might break the existing API.

Reply:
> Maybe the `home_page` should also expose the channel trailer, but that might break the existing API.

The old channel parser was implemented quite recently, so you shouldn't worry about backward compatibility here.